res/controllers/common-controller-scripts.js

```diff
@@ -1,128 +1,169 @@
 // Functions common to all controllers go in this file
+/* global print:off
+       printObject:off
+       stringifyObject:off
+       arrayContains:off
+       secondstominutes:off
+       msecondstominutes:off
+       colorCodeToObject:off
+       colorCodeFromObject:off
+       script:off
+       bpm:off
+       ButtonState:off
+       LedState:off
+       Controller:off
+       Button:off
+       Control:off
+       Deck:off
+ */
 
 // ----------------- Prototype enhancements ---------------------
 
 // Returns an ASCII byte array for the string
 String.prototype.toInt = function() {
-    var a = new Array()
+    var a = new Array();
     for (var i = 0; i < this.length; i++) {
-        a[i] = this.charCodeAt(i)
+        a[i] = this.charCodeAt(i);
     }
-    return a
-}
+    return a;
+};
 
 // ----------------- Function overloads ---------------------
 
 // Causes script print() calls to appear in the log file as well
-print = function(string) {
-    engine.log(string)
-}
+var print = function(string) {
+    engine.log(string);
+};
 
-printObject = function(obj, maxdepth) {
-    print(stringifyObject(obj, maxdepth))
-}
+// eslint-disable-next-line no-unused-vars
+var printObject = function(obj, maxdepth) {
+    print(stringifyObject(obj, maxdepth));
+};
 
-stringifyObject = function(obj, maxdepth, checked, prefix) {
+var stringifyObject = function(obj, maxdepth, checked, prefix) {
```

Comment:
it would be really nice to take this js cleanup and put it in a separate PR from the rest of the change.  mixing cleanup with code changes is really difficult to read, and they separate really nicely.

Reply:
Ok, I created a new branch from `master` and cherry picked the two cleanup commits. Please have a look at #2544. I'll remove the commits here when that one has been merged.